File: apps/problems/word-search/boilerplate-full/function.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include <sstream>
#include <climits>

//##USER_CODE_HERE##


int main() {
    int m_board;
std::cin >> m_board;
int n_board;
std::cin >> n_board;
std::vector<std::vector<char>> board(m_board, std::vector<char>(n_board));

for(int i = 0; i < m_board; ++i) {
    for(int j = 0; j < n_board; ++j) {
        std::cin >> board[i][j];
    }
}
std::string word;
std::cin >> word;
    Solution obj;
    bool result = obj.exist(board, word);
    std::cout << (result ? "true" : "false") << std::endl;
    return 0;
}

File: apps/problems/word-search/boilerplate-full/function.java

    import java.io.*;
    import java.util.*;

    

    //##USER_CODE_HERE##

    public class Main {
      
    public static void main(String[] args) throws IOException {
      BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
      int rows_board = Integer.parseInt(br.readLine().trim());
        int cols_board = Integer.parseInt(br.readLine().trim());
        char[][] board = new char[rows_board][cols_board];
        for (int i = 0; i < rows_board; i++) {
          String line = br.readLine().trim().replace(" ", "");
          for (int j = 0; j < cols_board; j++) {
            board[i][j] = line.charAt(j);
          }
        }
String word = br.readLine().trim();
      Solution obj = new Solution();
      boolean result = obj.exist(board, word);
      System.out.println(result);
      br.close();
    }
      
    }
  